src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -44,16 +44,30 @@
         }                                                               \
     } while (0)
 
+#ifdef HAVE_ERROR_CHECKING
 #define MPIDI_OFI_GET_BASIC_TYPE(a,b)   \
     do {                                        \
         if (MPIR_DATATYPE_IS_PREDEFINED(a))     \
             b = a;                              \
         else {                                  \
             MPIR_Datatype *dt_ptr;              \
             MPIR_Datatype_get_ptr(a,dt_ptr);    \
+            MPIR_Datatype_valid_ptr(dt_ptr, mpi_errno); \
             b = dt_ptr->basic_type;             \
         }                                       \
     } while (0)
+#else
+#define MPIDI_OFI_GET_BASIC_TYPE(a,b)           \
+    do {                                        \
+        if (MPIR_DATATYPE_IS_PREDEFINED(a))     \
+            b = a;                              \
+        else {                                  \
+            MPIR_Datatype *dt_ptr;              \
+            MPIR_Datatype_get_ptr(a,dt_ptr);    \
+            b = dt_ptr->basic_type;             \
+        }                                       \
+    } while (0)
+#endif
```

Comment:
This is uncomfortable to read. All we want is check valid datatype pointer without return error? I would suggest expand `MPIR_Datatype_valid_ptr` or create a custom macro for it here. That way the code will read as their intention.

Reply:
I don't think expanding `MPIR_Datatype_valid_ptr` will help this situation since we'd still have to add that check everywhere that we currently use `MPIDI_OFI_GET_BASIC_TYPE`.

I believe the reason I did it this way is because you can't have an `#ifdef` inside a `#define`.

Reply:
In `mpir_datatype.h` we added `MPIR_Assert(dt_ptr)` statements I think to address similar issues. Maybe we can use that here instead and avoid the duplicate macro definition?

Reply:
Done